lib/mutant/ast/regexp.rb

```diff
@@ -0,0 +1,41 @@
+module Mutant
+  module AST
+    # Regexp source mapper
+    module Regexp
+      UNSUPPORTED_EXPRESSION_TYPE = :conditional
+      private_constant(*constants(false))
+
+      # Check if expression is supported by mapper
+      #
+      # @param expression [Regexp::Expression]
+      #
+      # @return [Boolean]
+      def self.supported?(expression)
+        return if expression.type.equal?(UNSUPPORTED_EXPRESSION_TYPE)
```

Comment:
What about changing it to something like:

```ruby
def self.supported?(expression)
  return true if expression.terminal?
  expression.all? do |subexp|
    !subexp.type.equal?(UNSUPPORTED_EXPRESSION_TYPE) &&
    public_send(__method__, subexp)
  end
end
```

This way it's more clear what is happening, and the method is [total](https://en.wikipedia.org/wiki/Partial_function#Total_function).